src/shogun/neuralnets/NeuralLayer.h

```diff
@@ -147,10 +148,9 @@ class CNeuralLayer : public CSGObject
 	/** Computes the activations of the neurons in this layer, results should 
 	 * be stored in m_activations. To be used only with input layers
 	 * 
-	 * @param inputs activations of the neurons in the 
-	 * previous layer, matrix of size previous_layer_num_neurons * batch_size
+	 * @param inputs dot features to store activations from
 	 */
-	virtual void compute_activations(SGMatrix<float64_t> inputs) { }
+	virtual void compute_activations(CFeatures* inputs) { }
```

Comment:
I think we rather need to have GPU dot features that can do dot product right in the GPU memory.

If we go with dotfeatures API we get at least two advantages:

1) We can train with datasets that doesn't fit into RAM and GPU memory (but still use it as a kind of cache)
2) We can use complex features (random fourier, homogeneous kernel mapping, random mapping, etc) that can be computed on the fly with less memory footprint. I think (with giant batches and deep nets) GPUing things that are done after computing input would significantly speed up the learning process anyway.

That's all motivated by this COFFIN thing: http://www.icml2010.org/papers/280.pdf. That's a really cool idea by @sonney2k that could be one of the main features of our neural nets